### DISCUSSION TITLE
Framework Modules

### DISCUSSION THREAD

**Discussion by lefig** [oqtane/oqtane.framework#3021] [General] [answered]
I was wondering whether its possible to "entend" the core platform by defining a new custom common module.

A module that could be reused my multiple common modules.

Such as receiving integration messages using Dapr, SignalR etc,

Or perhaps as reuseable building blocks.

Thanks in advance

Fig

**Comment by sbwalker** [comment]
@lefig are you referring to a module or a Blazor component?

**Comment by lefig** [comment]
Thanks @sbwalker, a module with an API that may or may not have a blazor user interface as such.

So that for example other modules my reuse its middleware features such as message broker, event sourcing as library modules.

**Comment by sbwalker** [comment, marked as answer]
If you want to create a module which only provides backend functionality, you will still need to create a full project with Client and Server... the Client would contain only a single Index.razor component (which does not need to contain any UI markup or functionality). Your Client project would also have a ModuleInfo.cs to provide metadata for your module. You could then package this module and deploy it into an Oqtane instance. You would not need to add an instance of this module to any page in a site - the backend services would be actived on startup - but the module will appear in the Module Management area so that you can view its metadata, uninstall it, etc...

**Reply by lefig** [reply]
Perfect, and thank you for the instructions.
